files/public.go

```diff
@@ -119,6 +119,28 @@ func (storage *PublishedStorage) RemoveDirs(path string, progress aptly.Progress
 	return os.RemoveAll(filepath)
 }
 
+// Cleans up PublishedStorage tree towards the root up to first non-empty directory
+func (storage *PublishedStorage) CleanupPublishTree(path string) {
+	/*
+		Say, `path` is "a/b/c/d", let's remove all empty dirs:
+		 storage.rootPath + "a/b/c/d"
+		 storage.rootPath + "a/b/c"
+		 storage.rootPath + "a/b"
+		 ... and so on, up to first non-empty directory (or any other os.Remove error)
+	*/
+	segments := strings.Split(path, "/")
+
+	for segmentsCount := len(segments); segmentsCount > 0; segmentsCount-- {
+		err := os.Remove(
```

Comment:
I insist on checking for emptiness before remove here :)

also, the the line breaks are not needed, one line please

Reply:
> I insist on checking for emptiness

feel free to add it with your commit then

> breaks are not needed

fixed